pkg/api/common.go

```diff
@@ -116,8 +119,10 @@ func ParseFlags(fs *flag.FlagSet, cfg *Config) *Config {
 	fs.StringVar(&cfg.Auth.BearerToken, "bearer-token", "", "Bearer token (JWT) used for web endpoint authentication. Disabled by default. Mutually exclusive with bearer-token-file and basic auth methods.")
 	fs.StringVar(&cfg.Auth.BearerTokenFile, "bearer-token-file", "", "Path of the file containing the bearer token (JWT) used for web endpoint authentication. Disabled by default. Mutually exclusive with bearer-token and basic auth methods.")
 
+	fs.StringVar(&cfg.PromscaleEnableFeatures, "enable-feature", "", "Enable a beta/experimental features as a comma-separated list. Currently the following values can be passed: tracing")
+
 	// PromQL configuration flags.
-	fs.StringVar(&cfg.EnableFeatures, "promql-enable-feature", "", "[EXPERIMENTAL] Enable optional PromQL features, separated by commas. These are disabled by default in Promscale's PromQL engine. "+
+	fs.StringVar(&cfg.PromQLEnableFeatures, "promql-enable-feature", "", "[EXPERIMENTAL] Enable optional PromQL features, separated by commas. These are disabled by default in Promscale's PromQL engine. "+
```

Comment:
I think that for simplicity of use, we should consider joining `--enable-feature` with `--promql-enable-feature`.

Reply:
So my thinking here was that `--promql-enable-feature` enables features of the PromQL engine, which we aren't in control of i.e. they aren't really configuration values for _promscale_. For that reason, I thought it would make sense to keep them separate.

Reply:
I get this and it makes sense from the developer point of view. However if we put ourselves in the user role, that separation stops making much sense as both categories are essentially experimental features of promscale.

Reply:
If I am not wrong, i think we decided to have only one option, `--enable-feature` flag only, which will also serve PromQL flags as all PromQL based enable features have a `promql` prefix.

Reply:
> I get this and it makes sense from the developer point of view. However if we put ourselves in the user role, that separation stops making much sense as both categories are essentially experimental features of promscale.

You're right, I agree. What it means though (and this might be stating the obvious) is that we as promscale become "owners" (or "managers") of all experimental features in PromQL (and whatever other third-party libs we depend on), as well as our own. This will result in a (probably marginal) increase in our maintenance burden.

> If I am not wrong, i think we decided to have only one option, --enable-feature flag only, which will also serve PromQL flags as all PromQL based enable features have a promql prefix.

That's possible, I wasn't aware of the `-promql-enable-feature` flag when we had that discussion, so I think some of it went over my head 😅

Reply:
So taking a look into this again. In our [documentation](https://github.com/timescale/promscale/blob/master/docs/cli.md#promql-engine-evaluation-flags) we segregate between "promQL" and "other" features. I fear that now merging these things together makes that no longer congruent. @cevian or @ramonguiu can you weigh in on this discussion please?

Reply:
I think we should combine into one --enable-feature flag and NOT segregate. @antekresic may also have an opinion here since he is working on cli refactoring.

Reply:
Definitely use one flag for all our feature enabling needs, PromQL or otherwise. I'm just unsure if we want to do it in the scope of this PR or leave it out for a future one. 

If its not too big of a lift, I would like it in here :smiley:

Reply:
I have added this in the most recent commit. PTAL.